### ISSUE TITLE
UriFormatException for Git 2.13.1.windows.2

### ISSUE THREAD

**Issue by garretwilson** [git-for-windows/git#1217]
Suddenly when I even try to do a simple `git pull` on repositories I've used for months, with Git version 2.13.1.windows.2 I get:

```
fatal: UriFormatExceptionencounter.
    actualUrl
Password for 'https://user@bitbucket.org':
```

And it affects my TortoiseGit installation as well, causing it to ask for a password.

After I enter the password in either case, the Git operation succeeds.

I'm using Windows 10 Pro 64-bit with Creator's Update.

**Comment by linquize** [comment]
What credential helper are you using?
`git config --get credential.helper`

**Comment by garretwilson** [comment]
`manager`

The Git Credential Manager for Windows (GCM), I believe.

Another system is using `_netrc`, and it doesn't display this issue.

**Comment by garretwilson** [comment]
Very odd. It's working fine on a GitHub URL in the form `https://github.com/example/repo.git`. But it's failing for a Bitbucket URL in the form `https://user@bitbucket.org/example/repo.git`.

**Comment by garretwilson** [comment]
In Windows Credential Manager, the Bitbucket credentials are showing as "User name: user@example.com". But for GitHub, it's showing as "User name: Personal Access Token". I don't know if difference is significant.

**Comment by garretwilson** [comment]
I removed the Bitbucket credentials from Windows Credential Manager. Then I tried `git pull` again; it failed with the same error, and I had to enter my credentials. This time it added back a new set of credentials to Windows Credential Manager; curiously it now says "User name: user" instead of "User name: user@example.com".

Nevertheless the same thing happens with subsequent pulls: an error message is displayed and I have to manually enter my credentials.

**Comment by garretwilson** [comment]
I've opened this same issue at https://github.com/Microsoft/Git-Credential-Manager-for-Windows/issues/454 in case the GCM team has any ideas.

**Comment by dscho** [comment]
> I've opened this same issue at Microsoft/Git-Credential-Manager-for-Windows#454 in case the GCM team has any ideas.

Or in case that this is really a Git Credential Manager issue. Which it is :smile:

**Comment by garretwilson** [comment]
Please reopen this issue.

Yes, as you can see from the discussion at https://github.com/Microsoft/Git-Credential-Manager-for-Windows/issues/454 , this did turn out to be a GCM manager.

But the bug has been fixed in GCM, and Git for Windows is still including the buggy version in the distribution of Git 2.13.1, which makes it a Git for Windows bug.

*Please release a new version of Git for Windows.* The current release is broken.

Yes, I know they tell me I could install a newer GCM over my current Git for Windows, or try to patch something, or do some research into workarounds, or compile the code from scratch, or whatever; but the fact remains that the current Git for Windows release is broken, and this bug is valid until you integrate the unbroken GCM into Git for Windows.

Do you have a estimate on when you'll be releasing the new Git for Windows?

**Comment by dscho** [comment]
> <b>Please release a new version of Git for Windows.</b>

Wow. While you were shouting at me, I was finishing up the multi-hour effort of releasing a new Git for Windows version. As if my day was not stressful enough.

**Comment by garretwilson** [comment]
> While you were shouting at me ...

I was aware that all-caps indicate shouting, but I had thought that bold text simply was a way of emphasizing text so that it stood out from the rest. In this case I was merely attempting to distinguish my actual request from the rest of the discussion in my reply.

Is "bold text means shouting" a new convention I'm not aware of? If so I apologize. I certainly didn't mean to be shouting, not even in my mind.

**Comment by garretwilson** [comment]
I just wanted to confirm that v2.13.2.windows.1 seems to have resolved the problem. The first time I ran it I got a dialog asking for my Atlassian credentials, and after that I can access the remote repository, both in Git for Windows and in TortoiseGit.

Thanks for putting out another build, @dscho . I only posted my earlier message because I had thought you weren't producing another build because the bug was in a dependency. No harm intended.

The buggy behavior was very irritating, and I appreciate your putting out a fix.
